content/kotlin/concepts/queue/queue.md

```diff
@@ -0,0 +1,54 @@
+---
+Title: 'Queue'
+Description: 'A Queue is a collection interface that offers additional operations for accessing or manipulating items at the head of the queue.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Collections'
+  - 'Queues'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+A **`Queue`** is an interface in Kotlin that represents a data structure that stores elements in a linear order, and follows the First In, First Out (FIFO) principle. This means that the first element added to the queue will be the first one to be removed.
+
+### Syntax
+
+The syntax for creating a `Queue` in Kotlin using the `LinkedList` class is as follows:
+
+```pseudo
+val queue = LinkedList<Type>()
+```
+
+Where `Type` is the type of elements that the `Queue` will hold, for example, to store integers, use `Int` as the type :
+
+```pseudo
+val queue = LinkedList<Int>()
+```
+
+The `Queue` interface, which is implemented by `LinkedList`, can also be used:
+
+```pseudo
+val queue = LinkedList<Type>().toQueue()
+```
+
+### Example
+
+```kotlin
+// Create a new queue
+val queue = LinkedList<Int>()
+
+// Add elements to the queue
+queue.add(1)
+queue.add(2)
+queue.add(3)
+
+// Remove elements from the queue
+val firstElement = queue.remove() // Returns 1
+val secondElement = queue.remove() // Returns 2
```

Comment:
Can we add an output block following the example + explanation?

Reply:
@dakshdeepHERE Please print out the queue so we can see what it looks like after these lines